General Definition Updates - 5/14/2025

robertgk2017 opened the pull request:
TES4 \ EFSH \ DATA - Fix OptionalFrom
TES4 \ SOUN - Add AfterLoad
TES4 \ TREE - Add missing Record Flag and update names to match CK
TES4 \ WATR - Add ByteColors Defaults and Missing Record Flag

TES4R \ CELL - Add wbSimpleRecords check for XLRL

FO3 \ WATR - Set Default Values and Adjust required sub-record handling
FO3 \ WTHR \ Cloud Textures - Fix default value handling

FO3/FNV \ CELL \ XCET - Confirmed as Unused.
FO3/FNV \ DOBJ \ DATA - Add FormID Signature checks
FO3/FNV \ WEAP \ DNAM - 2 Fields confirmed as Unused

FO3/FNV/TES5/FO4/FO76 \ IMAD - Adjusted Defs, Structs, and set Required where applicable from the CKs

FO3/FNV/TES5/FO4/FO76/SF1 \ IDLM/PACK - Set Idle Animation Data to a common struct

TES5 - Add Detection for Community Shaders and new Flags it adds to CELL\LIGH.
TES5 - Make legacy OBScript Sub-Records un-editable
TES5 \ wbObjectTypeEnum - Move Enum above its uses, and update its values
TES5 \ CELL/WRLD  \ XWEM - Added DontShows based on the Game/CK's Requirements
TES5 \ DOBJ - Update handling and Sort Sig list
TES5 \ EFSH \ DATA - Decoding from Cobb
TES5 \ MOVT \ SPED/INAM - Update Float Handling
TES5 \ PACK - General Cleanup
TES5 \ PACK \ OnBegin/OnEnd/OnChange - change wbPDTOs to wbPDTO
TES5 \ PACK \ Package Data - Update formatting/collapse wbTargetData Variable
TES5 \ PACK \ PSDT - Define Enums for Month and Minute, and add Range Checks for Hour and Date
TES5 \ PACK \ PSDT \ Duration - Add wbDiv formatter to match value to the CK
TES5 \ PACK \ XNAM - Set Size to 1 Byte
TES5 \ QUST \ Aliases \ Find Matching Reference From Event - Add QuestEvent and EventMember Enums
TES5 \ REGN \ Region Data Entries - Add dont show to wbICON
TES5 \ RGDL - Initial implementation copying from FO3/FNV Defs.  Needs further checking from Cobb
TES5 \ SCEN \ Actors \ ALID - Add AliasToStr to resolve the Alias IDs
TES5 \ SCRLL/SPEL \ SPIT - Split handling to match differences in the CK.
TES5 \ SNDR - Update FNAM/LNAM Handling
TES5 \ SNDR \ GNAM - Add FormID Signature check
TES5 \ SOPM - Updated handling from Cobb
TES5 \ WATR - Clean Up, Default Values, Set Required according to the CK
TES5 \ WTHR \ NAM2/NAM3 - These are actually Sun/Moon Glare for version 35, before they got moved to NAM0

TES5/FO4/FO76 \ MGEF \ SNDD - Add Alignment Sort Key
TES5/FO4/FO76 \ STAT - Add handling to enforce setting the Has Distant LOD flag, for adding/setting LOD Models

TES5/FO4/FO76/SF1 - Update wbEvenetMemberEnum handling
TES5/FO4/FO76/SF1 \ wbConditions - Fix Alignment Sort Keys for CIS1/CIS2
TES5/FO4/FO76/SF1 \ wbConditions \ CTDA \ Parameter #3 - Partially Decoded as a Union relative to the `Run On` Enum
TES5/FO4/FO76/SF1 \ ADDN/COLL - Add NamedIndex Support
TES5/FO4/FO76/SF1 \ LCTN - Add Sortkeys and set cpBenign on Data Table Arrays
TES5/FO4/FO76/SF1 \ NAVI \ NVMI - Category changed to Flags
TES5/FO4/FO76/SF1 \ SCEN - Housekeeping on all SCEN Defs, move more dentical duplicate functions to Common

FO4 / NAVI / NVMI / Category - Define Unknown 96 so it doesn't throw errors

FO4/FO76/SF1 \ wbAPPR - Make Sorted, confirmed order does not matter in any of the games by Eck

FO76 \ wbConditions - Parameter QuestStageToStr's don't target Objectives, only Stages.

SF1 - Preliminary Reflection Support, Editing is still hard blocked.  Do not try to enable it.
SF1 \ ReferenceRecord - Added XALG Sub-Record
SF1 \ wbHNAMHNAM - Rename to wbHeadtracking
SF1 \ PNDT - Removed Required from CNAM, Decoded EOVR \ Worldspace \ [2], Add NULL to INAM \ Atmosphere Valid Sigs
SF1 \ PNDT \ CNAM/EOVR - Added AfterLoad for proper handling.
SF1 \ QUST \ Aliases \ ALCO - Add Project to list of valid FormIDs

ALL \ MSTT \ DATA - Updated handling/Decoded for FO3/FNV
ALL \ NAVI - Decoding of Navmesh Bounds and Approx Location
ALL \ NAVM - Labeling of Navmesh Bounds

Common - Split To String Callback functions and procedures into 2 Groups.
Common \ wbByteColors - Add ability to set Default values
Common \ wbMagicEffectSounds - Set wbArray to Sorted, doesn't affect game and CK will write in random orders.
Common \ wbNavmeshCoverFlags - Starfield has an additional flag.
Common \ wbSCENAddInfo - Fix Sub-Record signature call

Implementation - Fix dfCanContainReflection checks

Moved the following functions to common:
wbAliasLinksTo
wbConditionAliasToStr
wbDOBJObjectsAfterLoad
wbPackageLocationAliasToStr
wbQuestAliasToStr
wbQuestExternalAliasToStr
wbSceneAliasToStr
wbScriptObjectAliasToStr
wbStrToAlias - Rename to wbAliasToInt

Script - Add TControl StyleEelements Options
Script - Add a ScaleValue function for TControls

Script \ Assets Manager - Add Scaling TControl Values
Script \ Find ESP Plugins which could be turned into ESL - Add ESM Check for New Cells warning
Script \ Weather Editor - Add Scaling to TControl Values
Script \ Weather Editor - Fix Color Panels not showing when Themes are applied

xEdit - Fix Not setting UDR'd references enable parent to Opposite State in TES4
xEdit - Fix Script window Line # and Col # reporting the wrong value
xEdit - Remove Asterisk from element compare filenames
xEdit - Temporarily disable QAC setting persistent flag on all UDR'd ACHR/ACREs.
xEdit \ Implementation \ TwbMainRecord.Init - Move AfterLoad Execution to After Required Sub-Record processing
xEdit \ Filter for Unecessary Persistent Refs - Add extra handling for cases of Required persistence
Implementation \ TwbMainRecord.Reached - Change DIAL/DLBR/SCEN handling to just check if their parent quest is Reachable

BSArch - Allow .fuz files to be compressed, they ship compressed from Bethesda.

NIF - Update handling of Apply Transforms/Update Bounds for Skinned BSTriShapes
NIF \ wbbhkCOFlags - Add missing flags
NIF \ wbSkyrimLayer - Add missing Collision Layers
NIF \ CalculateTangentsBitanges2 - Fix NaN handling
NIF \ UpdateNormals - Fix Delphi short circuit evaluation
NIF \ Skyrim - Update Shader Flags/Type Names

SNIFF - Move MSHeap to first Uses unit in project file
SNIFF - Update to version 1.7, Re-Organize file structure, Add to GroupProject

SNIFF \ Check For Errors \ CheckAllWhiteVertexColors - Skip Parallax shader types.
SNIFF \ Check for Errors \ CheckBSXFlags - Report uneeded BSXFlags
SNIFF \ Check For Errors \ CheckMiscellaneous - Check BSEffect/LightingShaderProperties for Opposite type of Controllers
SNIFF \ Check For Errors \ CheckMiscellaneous - Add check for BSTriShapes in facegen nifs, should be BSDynamicTriShape
SNIFF \ Check For Errors \ CheckInvalidRepeatedChildrenNames - Made string comparisons Case Sensitive
SNIFF \ Check For Errors \ CheckParticleSystems - Check NiParticlesData \ Num Subtexture Offsets is greater than 16
SNIFF \ Check For Errors \ CheckParticleSystem - Num Subtexture Offsets limit increased to 256 in Skyrim
SNIFF \ Check For Errors \ CheckShaderTypeFlags - Update and Expand checks for missing Shader Flags and Textures
SNIFF \ Check For Errors \ CheckSkinningIssues - Add check for BSDismemberSkinInstance\Num Partitions is less than NiSkinPartitions
SNIFF \ Check For Errors \ CheckTextureSetSlots - Add checks/report for textures being set that aren't used

SNIFF \ Vertex Color Painting \ Remove - Skip Parallax and Tree_Anim shaders in Skyrim meshes

SNIFF \ Universal Fixer \ FixBSXFlags - Remove empty BSXFlags
SNIFF \ Universal Fixer \ FixCollision - Update Motion System/Mass/Tensor handling for various layers
SNIFF \ Universal Fixer \ FixShaderProperty - Add MultiLayer Parallax Shader Type/Flag checks.
SNIFF \ Universal Fixer \ FixShaderProperty - Update Own Emit shader flag handling
SNIFF \ Universal Fixer \ FixShaderProperty - Add Specular shader flag handling

robertgk2017 committed:
Script - Add TControl StyleEelements Options

robertgk2017 committed:
Script \ Weather Editor - Fix Color Panels not showing when Themes are applied

robertgk2017 committed:
TES5/FO4/FO76 \ MGEF \ SNDD - Add Alignment Sort Key

zilav commented on xEdit/JvI/xejviScriptAdapterMisc.pas:
```diff
@@ -2176,6 +2188,13 @@ procedure RegisterJvInterpreterAdapter(JvInterpreterAdapter: TJvInterpreterAdapt
+    { TControl }
+    AddConst('Controls', 'seFont', Ord(seFont));
+    AddConst('Controls', 'seClient', Ord(seClient));
+    AddConst('Controls', 'seBorders', Ord(seBorder));
```
Looks like you also fell victim of seBorder(s)

robertgk2017 replied:
Typos are our friends!

robertgk2017 committed:
I love typos.

robertgk2017 committed:
xEdit - Remove Asterisk from element compare filenames

jbostrus committed:
Fix external compare button failing if tab is detected as modified.

robertgk2017 committed:
Revert "xEdit - Remove Asterisk from element compare filenames"

This reverts commit 8a27a6f37dc7351fa56338357b73b04b5c55cb91.

robertgk2017 committed:
Merge pull request #15 from jbostrus/Misc

Fix external compare button failing if tab is detected as modified.

robertgk2017 committed:
Script - Add a ScaleValue function for TControls

robertgk2017 committed:
Script \ Assets Manager - Add Scaling TControl Values

robertgk2017 committed:
Scripts \ Weather Editor - Add Scaling to TControl Values

robertgk2017 committed:
SF1 \ PNDT - Removed Required from CNAM, Decoded EOVR \ Worldspace \ [2], Add NULL to INAM \ Atmosphere Valid Sigs

robertgk2017 committed:
TES5 \ SNDR - Update FNAM/LNAM Handling

Kuroitsune committed:
FaceGen Standardization

Decoded FGGS/FGGA/FGTS for TES4, FO3, FNV.

Note: The sliders in the official tools simply amplify the effect, the actual data stored are just the movement of the morph keys. The EGM/EGT store the associated rigging.

One day I'll decode them. Someday that is not any time soon.

This time I didn't accidently revert it. Also, changed the collapse flag to "Other."

robertgk2017 committed:
Merge branch 'pr/1432' into misc

robertgk2017 committed:
Common - Move wbFacegen

robertgk2017 committed:
TES4R \ CELL - Add wbSimpleRecords check for XLRL

robertgk2017 committed:
TES5 \ SOPM - Updated handling from Cobb

robertgk2017 committed:
SNIFF - Update to version 1.7, Re-Organize file structure, Add to GroupProject

robertgk2017 committed:
Forgot to edit the temp file path

robertgk2017 committed:
SNIFF - Move Proc units to Sub-Folder

robertgk2017 committed:
Common \ wbNavmeshCoverFlags - Starfield has an additional flag.

robertgk2017 committed:
SF1 \ ReferenceRecord - Added XALG Sub-Record

robertgk2017 committed:
SF1 \ QUST \ Aliases \ ALCO - Add Project to list of valid FormIDs

robertgk2017 committed:
TES5 - Add Detection for Community Shaders and new Flags it adds to CELL\LIGH.

robertgk2017 committed:
xeMainForm - Temporarily disable QAC setting persistent flag on all UDR'd ACHR/ACREs.  Note the comment.

robertgk2017 committed:
TES5 \ PACK \ XNAM - Set Size to 1 Byte

robertgk2017 committed:
xEdit - Fix Not setting UDR'd references enable parent to Opposite State in TES4

robertgk2017 committed:
SNIFF - Move MSHeap to first Uses unit in project file

robertgk2017 committed:
TES5 \ MOVT \ SPED/INAM - Update Float Handling

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixShaderProperty - Add MultiLayer Parallax Shader Type/Flag checks.

robertgk2017 committed:
And forgot the scaling

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckInvalidRepeatedChildrenNames - Check against String Index instead of String Text

robertgk2017 committed:
Revert "SNIFF \ Check For Errors \ CheckInvalidRepeatedChildrenNames - Check against String Index instead of String Text"

This reverts commit 1326020d3b4214e5d651cd0e52b915e32f297ac8.

robertgk2017 committed:
Whoops

robertgk2017 committed:
ALL \ NAVI - Decoding of Navmesh Bounds and Approx Location

robertgk2017 committed:
ALL \ NAVM - Labeling of Navmesh Bounds

robertgk2017 committed:
FO3/FNV \ CELL \ XCET - Confirmed as Unused.

robertgk2017 committed:
FO3/FNV \ WEAP \ DNAM - 2 Fields confirmed as Unused

robertgk2017 committed:
ALL \ MSTT \ DATA - Updated handling/Decoded for FO3/FNV

robertgk2017 committed:
Implementation \ TwbMainRecord.Init - Move AfterLoad Execution to After Required Sub-Record processing

robertgk2017 committed:
TES5 \ SNDR \ GNAM - Add FormID Signature check

robertgk2017 committed:
Nope

robertgk2017 committed:
Oops

robertgk2017 committed:
FO3 \ WTHR \ Cloud Textures - Fix default value handling

robertgk2017 committed:
Common \ wbByteColors - Add ability to set Default values

robertgk2017 committed:
FO3 \ WATR - Set Default Values and Adjust required sub-record handling

robertgk2017 committed:
TES4 \ WATR - Add ByteColors Defaults and Missing Record Flag

robertgk2017 committed:
TES4 \ TREE - Add missing Record Flag and update names to match CK

robertgk2017 committed:
TES4 \ SOUN - Add AfterLoad

robertgk2017 committed:
Misc wbUnuseds

robertgk2017 committed:
Fix NAVI Summaries

robertgk2017 committed:
And 2 more

robertgk2017 committed:
TES4 \ EFSH \ DATA - Fix OptionalFrom

robertgk2017 committed:
FO3/FNV \ DOBJ \ DATA - Add FormID Signature checks

robertgk2017 committed:
Move wbDOBJObjectsAfterLoad to Common

robertgk2017 committed:
SF1 \ PNDT \ CNAM/EOVR - Added AfterLoad for proper handling.

robertgk2017 committed:
ALL \ wbConditions \ Function - Remove UpdateSameParentUnions

robertgk2017 committed:
TES5 \ DOBJ - Update handling and Sort Sig list

robertgk2017 committed:
NIF \ wbSkyrimLayer - Add missing Collision Layers

robertgk2017 committed:
NIF \ wbbhkCOFlags - Add missing flags

robertgk2017 committed:
SNIFF \ Check For Errors - Check BSEffect/LightingShaderProperties for Opposite type of Controllers

robertgk2017 committed:
NIF - Fix NaN handling for Normals/Tangents

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixShaderProperty -  Add handling for Adding/Removing Own_Emit and EnvMap_Light_Fade shader flags depending on Shader Type

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckSkinningIssues - Add check for BSDismemberSkinInstance\Num Partitions lower than NiSkinPartition

robertgk2017 committed:
OOPS

robertgk2017 committed:
This time it works I swear

robertgk2017 committed:
FO4 / NAVI / NVMI / Category - Define Unknown 96 so it doesn't throw errors

robertgk2017 committed:
TES5 \ EFSH \ DATA - Decoding from Cobb

robertgk2017 committed:
SNIFF \ Universal Fixer \ Fix Shader Properties - Update Own Emit shader flag handling

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixShaderProperty - Add Specular shader flag handling

robertgk2017 committed:
TES5 \ EFSH \ DATA - Update flags, again from Cobb

robertgk2017 committed:
Fix 2 compiler warnings

robertgk2017 committed:
Now this wants to work

robertgk2017 committed:
TES5 \ RGDL - Initial implementation copying from FO3/FNV Defs.  Needs further checking from Cobb

robertgk2017 committed:
i love typos

robertgk2017 committed:
NIF \ UpdateNormals - Fix Delphi short circuit evaluation

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ NAVI \ NVMI - Category changed to Flags

robertgk2017 committed:
xEdit \ Filter for Unecessary Persistent Refs - Add extra handling for cases of Required persistence

robertgk2017 committed:
FO3/FNV/TES5/FO4/FO76 \ IMAD - Adjusted Defs, Structs, and set Required where applicable from the CKs

robertgk2017 committed:
NIF - Update handling of Apply Transforms/Update Bounds for Skinned BSTriShapes

robertgk2017 committed:
FO3/FNV/TES5/FO4/FO76/SF1 \ IDLM/PACK - Set Idle Animation Data to a common struct

robertgk2017 committed:
Common \ wbMagicEffectSounds - Set wbArray to Sorted, doesn't affect game and CK will write in random orders.

robertgk2017 committed:
TES5 \ PACK - General Cleanup

jbostrus committed:
Feature: Added support when using the Add menu item on a record group to hold shift and add multiple at one time.

robertgk2017 committed:
Merge pull request #16 from jbostrus/Misc

Feature: Added support when using the Add menu item on a record group…

robertgk2017 committed:
TES5 \ SCEN \ Actors \ ALID - Add AliasToStr to resolve the Alias IDs

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckShaderTypeFlags - Update and Expand checks for missing Textures

robertgk2017 committed:
Todd made me do it!

robertgk2017 committed:
TES5/FO4/FO76/SF1 - Update wbEvenetMemberEnum handlind

robertgk2017 committed:
TES5 \ QUST \ Aliases \ Find Matching Reference From Event - Add QuestEvent and EventMember Enums

robertgk2017 committed:
NIF \ Skyrim - Update Shader Flags/Type Names

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ LCTN - Add Sortkeys and set cpBenign on Data Table Arrays

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckShaderTypeFlags - Update and Expand missing shader flag and texture checks

robertgk2017 committed:
TES5 \ wbObjectTypeEnum - Move Enum above its uses, and update its values

robertgk2017 committed:
TES5 \ PACK \ Package Data - Update formatting/collapse wbTargetData Variable

robertgk2017 committed:
TES5 \ PACK \ PSDT \ Duration - Add wbDiv formatter to match value to the CK

robertgk2017 committed:
TES5 \ PACK \ PSDT - Define Enums for Month and Minute, and add Range Checks for Hour and Date

robertgk2017 committed:
TES5 \ wbConditions \ Parameter #3 - Add Event Member Enum

robertgk2017 committed:
Typos are my friends

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckBlocksOrder - Skip bhkBallSocketConstraintChain

robertgk2017 committed:
Update afkmods link.

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckShaderTypeFlags - Setup seperated check for BSLighting/BSEffect ShaderProperties

robertgk2017 committed:
Can't make this stuff up, I swear

eckserah committed:
Add changes for latest PTS

Add new condition function HasCAMPWeatherActive
New entry points Mod Projectile Bounce Count, Mod NPC Normalized Min Level, Mod NPC Normalized Max level, Mod NPC Normalized Level, Mod Ammo Spender Max Stack Count, Mod Ammo Spender Max Reload Stack Mult
Add CSTD for CSTY
Add FIEX, FISD, FISU for FISH
Add LVUD for LVLI

robertgk2017 committed:
Fix LCTN sub-record order.

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckTextureSetSlots - Add checks/report for textures being set that aren't used

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckMiscellaneous - Add check for BSTriShapes in facegen nifs, should be BSDynamicTriShape

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckAllWhiteVertexColors - Skip Parallax shader types.

robertgk2017 committed:
Common \ wbWorldAfterLoad - Fix if check

robertgk2017 committed:
TES5 - Make legacy OBScript Sub-Records un-editable

robertgk2017 committed:
TES5 \ PACK \ OnBegin/OnEnd/OnChange - change wbPDTOs to wbPDTO

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ wbConditions \ CTDA \ Parameter #3 - Partially Decoded as a Union relative to the `Run On` Enum

robertgk2017 committed:
Move wbConditionAliasToStr to Common

robertgk2017 committed:
Common - Split To String Callback functions and procedures into 2 Groups.

robertgk2017 committed:
FO76 \ wbConditions - Parameter QuestStageToStr's don't target Objectives, only Stages.

robertgk2017 committed:
Move wbScriptObjectAliasToStr to Common

robertgk2017 committed:
Oops

robertgk2017 committed:
Move wbPackageLocationAliasToStr to Common

robertgk2017 committed:
Move wbQuestAliasToStr to Common

robertgk2017 committed:
Move wbAliasLinksTo to Common

robertgk2017 committed:
Move wbStrToAlias to Common, rename to wbAliasToInt

robertgk2017 committed:
Move wbQuestExternalAliasToStr to Common

robertgk2017 committed:
Move wbSceneAliasToStr to Common

robertgk2017 committed:
SF1 - Rename wbHNAMHNAM to wbHeadtracking

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ SCEN - Housekeeping on all SCEN Defs, move more dentical duplicate functions to Common

robertgk2017 committed:
TES5 - Collapse wbEquipType variable

robertgk2017 committed:
It's all Todds fault...

robertgk2017 committed:
TES5/FO4/FO76 \ STAT - Add handling to enforce setting the Has Distant LOD flag, for adding/setting LOD Models

robertgk2017 committed:
xEdit - fix Script window Line # and Col # reporting the wrong value

robertgk2017 committed:
Remove few unused variables

robertgk2017 committed:
Typo

robertgk2017 committed:
SNIFF \ Vertex Color Painting \ Remove - Skip Parallax and Tree_Anim shaders in Skyrim meshes

robertgk2017 committed:
*Terrified Lizard Noises*

robertgk2017 committed:
xEdit - Prevent file saving if the "Optimized" File Flag is set.

Bobbyclue committed:
SF1 and TES5 definition updates

TES5 - Allowed SOUN to be used in place of SNDR. Confirmed to work in game.
TES5 - Added VMAD to ALCH, STAT, and SCOL
TES5 - Renamed various unknown flags on magic items to match other magic items as these flags are shared
TES5 \ MUSC \ FNAM - Renamed flag 'Unknown 4' to 'Maintain Track Order'
TES5 - Ensured all leveled form flags and LVLG are present across the various types. Confirmed to work in game.

SF1 - Added TACT and LVLB to list of placeable references
SF1 - Defined DAMC \ wbUnknown(4) as a CURV
SF1 - Noted additional component types not yet defined
SF1 - Renamed ENAM in Animation Graph Components to 'Controls Base Skeleton'
SF1 - Renamed ENAM in Animation Graph Components to 'Controls Base Skeleton'
SF1 - Allowed NULL in BNAM under Worldspace Overlay Components
SF1 - Added generic list of form flags that are present on most records to various forms, per the inspector window in the CK
SF1 - Added wbBaseFormComponents to  various forms that lacked them previously
SF1 \ ACTI - Corrected order of MWAT and ATTX
SF1 \ TACT - Added additional subrecord types, per the inspector window and in game testing
SF1 \ REFR - Added wbBaseFormComponents and wbUnknown(XLTW)
SF1 \ LCRT, AACT,  - Added various subrecords added via the inspector window (identical to keywords)
SF1 \ HAZD  - Added various subrecords added via the inspector window (identical to keywords)
SF1 \ NAVI - Added handful of flags and valid form inputs that occur in the vanilla game (no, not the horribly broken ones)
SF1 \ MUSC \ FNAM - Renamed '' flag to 'Removal Queued'
SF1 \ INFO - Defined TROI
SF1 \ INGR - Reenabled ingredient support now that the CK does not crash when loading them
SF1 \ KEYM - Added additional subrecords created via the inspector window
SF1 \ LVSP - Added additional subrecords created via the inspector window
SF1 \ MISC - Added additional subrecords created via the inspector window
SF1 \ COBJ - Added additional subrecords created via the inspector window
SF1 - Defined instances of ALFG as the float 'Orbit Altitude Mult'
SF1 - Allowed ALCO to create additional form types ALCH, LVLB, and LVLN
SF1 - Removed SetRequired from instances of ALAM, the CK does not fill this value always
SF1 \ QUST \ QDUP - Added GLOB, SCEN, and QUST to the list of valid forms
SF1 \ RACE - Added XALG
SF1 \ ACTR - Added additional subrecords, often shared with REFR
SF1 \ REFR \ XLFD - Added additional unknown flag that the CK can add
SF1 - Defined some unknown SPIT flags
SF1 - Defined SCRL

robertgk2017 committed:
Merge pull request #17 from Bobbyclue/misc

SF1 and TES5 definition updates

robertgk2017 committed:
Oops

robertgk2017 committed:
Cleanup

robertgk2017 committed:
This isn't supported either.

Bobbyclue committed:
SF1 definition updates

SF1 - Added handling for mod added terminal backgrounds matched to how custom map marker types are handled in prior games
SF1 - Renamed 'Texture Overlay' to 'Texture' in wbPhotoModeEnum to better match the CK
SF1 - Added missing keyword type, 'AVMS Condition Keyword', to wbKeywordTypeEnum and renamed some types to better match the CK
SF1 \ AMMO - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ AMMO - Corrected order of wbDEST
SF1 \ LAYR - Added general set of form flags present on most forms. Added via inspector window
SF1 \ LENS - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ LGDI - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ NOTE - Added general set of form flags present on most placeable forms. Added via inspector window
SF1 \ LENS - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ TRNS - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ TRNS - Defined some default values and required subrecords to better align with the CK
SF1 \ AAPD - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ AAMD - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ AAMD - Defined default values to better match the CK
SF1 \ SECH - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ CURV - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ STND - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ STMP - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ GCVR - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ GCVR - Defined default editor values to better match the CK
SF1 \ VOLI - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ PMFT - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ CHAL - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ FXPD - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ CNDF - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ AOPF - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ PDCL - Added general set of form flags present on most placeable forms. Added via inspector window
SF1 \ CUR3 - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ BMMO - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ GBFT - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ LVLB - Added general set of form flags and subrecords present on most placeable forms. Added via inspector window
SF1 \ MAAM - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ MAAM - Defined default values to better match the CK
SF1 \ IRES - Added general set of form flags present on most forms. Added via inspector window
SF1 \ AOPS - Added general set of form flags present on most forms. Added via inspector window
SF1 \ AOPS - Defined default values to better match the CK
SF1 \ ATMO - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ BMOD - Added general set of form flags present on most forms. Added via inspector window
SF1 \ CLDF - Added general set of form flags present on most forms. Added via inspector window
SF1 \ EFSQ - Added general set of form flags present on most forms. Added via inspector window
SF1 \ FORC - Added general set of form flags present on most forms. Added via inspector window
SF1 \ LVSC - Added general set of form flags present on most forms. Added via inspector window
SF1 \ MRPH - Added general set of form flags present on most forms. Added via inspector window
SF1 \ MTPT - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ OSWP - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ PCMT - Added general set of form flags and subrecords present on most forms. Added via inspector window
SF1 \ PSDC - Renamed from 'Particle System Define Collision' to 'Particle System Define Collection' to match the CK

robertgk2017 committed:
Script - Add ESM Check for New Cells warning in the Find ESL Capable plugins script

robertgk2017 committed:
Merge pull request #18 from Bobbyclue/misc

SF1 definition updates - Flags 'n Stuff

robertgk2017 committed:
Forgot this

robertgk2017 committed:
Common \ wbLCTNCellDontShow - Fix Handling

robertgk2017 committed:
SNIFF \ Check For Erros - Check NiParticlesData \ Num Subtexture Offsets is greater than 16

robertgk2017 committed:
Unecessary Begin/Ends :)

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckParticleSystem - Num Subtexture Offsets limit increased to 256 in Skyrim

robertgk2017 committed:
I was just kidding

robertgk2017 committed:
You get a Typo, and YOU get a typo, Everyone gets a Typo.

robertgk2017 committed:
SNIFF \ Check for Errors \ Report uneeded BSXFlags

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixBSXFlags - Remove empty BSXFlags

robertgk2017 committed:
Yup

Bobbyclue committed:
SF1 Definition Updates

SF1 \ AMMO - Reverted unintended change that could cause FLLD to not load correctly
SF1 \ LCTN - Marked various unknown pieces as benign conflicts.
SF1 \ LCTN - Defined additional form flags
SF1 \ LCTN - Added wbBaseFormComponents
SF1 \ LCTN - Gave ANAM a default value to match the CK
SF1 \ CLFM - Defined additional form flags
SF1 \ CLFM - Added FULL, can be added through the inspector window
SF1 \ LSCR - Added ICON, NNAM, TNAM, ONAM, ZNAM, and MOD2. Can be added via the inspector window
SF1 \ REFR \ XLLD - No longer treated as a bool enum as additional data can be contained under certain circumstances
SF1 \ WRLD - Added additional form flags

robertgk2017 committed:
That was on Purpose i swear

robertgk2017 committed:
Merge pull request #19 from Bobbyclue/misc

SF1 Definition Updates

robertgk2017 committed:
TES5 \ PACK \ PSDT \ Minute - Quick fix for Enum sorting

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixCollision \ Update Motion System/Mass/Tensor handling for various layers

robertgk2017 committed:
TES5 \ SCRLL/SPEL \ SPIT - Split handling to match differences in the CK.

robertgk2017 committed:
oops

Bobbyclue committed:
FO4 and SF1 Definition Updates

SF1 - Added VMAD and BaseFormComponents to various records that support them (accessible through inspector window and papyrus component)
FO4 - Removed unused wbWeaponAnimTypeEnum
FO4 \ REFR \ XPWR - Allowed ACHR records in addition to REFR (Can happen when swapping base object in CK)
FO4 \ REFR \ XASP - Renamed to Acoustic Parent
FO4 \ REFR \ XATP - Defined as Activation Point
FO4 \ REFR \ XCVR - Water Current Zone Data
FO4 \ ACHR - Defined wbActionFlag (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XLOC (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XLTW (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XPWR (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XATR (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XWCN (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined XWCU (Can be added when swapping base object in CK)
FO4 \ ACHR - Defined ONAM (Can be added when swapping base object in CK)
FO4 - Added additional form types to wbFormTypeEnum
FO4 - Defined various default values and required subrecords
FO4 \ AACT \ TNAM - Defined as Type
FO4 \ LCRT \ TNAM - Defined as Type
FO4 \ PACK - Updated schedule data to act as enums, as SSE is handled
FO4 \ PACK \ TPIC - Defined as Dialogue Topic
FO4 \ FLOR \ FNAM - Defined as Flags

robertgk2017 committed:
Implementation \ TwbMainRecord.Reached - Change DIAL/DLBR/SCEN handling to just check if their parent quest is Reachable

robertgk2017 committed:
Merge pull request #20 from Bobbyclue/misc

FO4 and SF1 Definition Updates

robertgk2017 committed:
TES5 \ WRLD \ XWEM - Fix Definition order

robertgk2017 committed:
TES5/FO4/FO76/SF1 \ wbConditions - Fix Alignment Sort Keys for CIS1/CIS2

robertgk2017 committed:
TES5 \ WATR - Clean Up, Default Values, Set Required according to the CK

robertgk2017 committed:
TES5 \ CELL/WRLD  \ XWEM - Added DontShows based on the Game/CK's Requirements

robertgk2017 committed:
TES5 \ WTHR - Define Old Cloud Textures

robertgk2017 committed:
TES5 \ WTHR - Minor cleanup

robertgk2017 committed:
TES5 \ NAM2/NAM3 - These are actually Sun/Moon Glare for version 35, before they got moved to NAM0

robertgk2017 committed:
Oops

robertgk2017 committed:
Common \ wbSCENAddInfo - Fix Sub-Record signature call

robertgk2017 committed:
Implementation - Fix dfCanContainReflection checks

robertgk2017 committed:
Update to 4.1.5p

robertgk2017 committed:
Remove unused commandline argument

robertgk2017 committed:
Well that clearly wasn't going to work

robertgk2017 committed:
SNIFF - Update version to 1.8

robertgk2017 committed:
Add missing Interface Bool

robertgk2017 committed:
Add new Signatures for Starfield

robertgk2017 committed:
Init - Add OBME Detection

robertgk2017 committed:
Make distinction between old and new WATR Noise Textures

robertgk2017 committed:
And another one

robertgk2017 committed:
SNIFF \ Check For Errors \ CheckShader - Reorganization

robertgk2017 committed:
SNIFF \ Universal Fixer \ FixShader - Reorganization

robertgk2017 committed:
Script \ Assets Manager - Literally all of the updates

robertgk2017 committed:
TES4 \ PACK \ PSDT \ Month - Update names to match CK

robertgk2017 committed:
Erroneous Semicolon

robertgk2017 committed:
SF1 - Preliminary Reflection Support, Editing is still hard blocked.  Do not try to enable it.

robertgk2017 committed:
BSArch - Allow .fuz files to be compressed, they ship compressed from Bethesda.

robertgk2017 committed:
Common \ wbPNDTAfterLoad - Fix exception.

robertgk2017 committed:
TES5 \ REGN \ Region Data Entries - Add dont show to wbICON

robertgk2017 committed:
FO4/FO76/SF1 \ wbAPPR - Make Sorted, confirmed order does not matter in any of the games by Eck

robertgk2017 committed:
TES5 \ REGN \ RGDS - Decode 2nd Unknown as a FormID pointing to an LTEX

robertgk2017 committed:
SF1 - More reflection handling

robertgk2017 committed:
TES5/FO4/FO7/SF1 \ ADDN/COLL - Add NamedIndex support

eckserah committed:
Updates for Live and PTS

Added FISH to the list of records allowed in certain condition functions
Remove second GetWeakPointDamageMultiplier
Add all known XALG flags
Add record types to record enum
Define all of FISH
Fix COBJ DNAM
Add Raid quest type
Add Cylinder primitive type for 76 and Starfield
Unknown 31 for weapon dnam flags is Has Overheating
Add Overheating rate values in DNAM

robertgk2017 committed:
Merge branch 'pr/1441' into misc
